### ISSUE TITLE
Suggestion about the Story Line

### ISSUE THREAD

**Issue by Phieldar** [olivierkes/manuskript#536]
I really love the software. I would like to propose something here about the Story Line. Would it be possible that the color line of plots in the story line have a editable beggining and end. Because there could be plots that start halfway in the project and end before the main plot ends... It would make the Story Line very useful to see the multiple beggining and ends of plots along the main story.
It wouls be useful in an academical project also since plots in fiction could be argument in an essay. In a long book, there are many argument... and big argument include smaller parts... and visualising it could be very helpful!

**Comment by worstje** [comment]
That is a pretty decent idea, I like it. 👍 

Maybe this idea could be refined so it isn't plot-specific, though. Maybe a simple checkbox on the timeline could work out in order to 'contextualize' all timelines. (Bad word no doubt, but I can't think of a better one.)

This would achieve that the timelines before the first reference of a character/plot would not be drawn. Similarly, the last reference would be where the lines end.

A while back I suggested in another issue (#476) that I would like a higher granularity so that references could link to a specific resolution step in a plot. Of course, stories being stories, some things don't always develop exactly in the order you planned and might instead slightly overlap. If such a thing were implemented, that could also be used to refine the way this feature behaved; with this setting turned on a timeline for a plot might only end if the reference also refers to the final resolution step, and if not, leaving the line drawn all the way to the end.

This would still leave things a bit awkward in regards to characters: what does a character timeline represent exactly? How to determine its end? Arguably you might want to see a characters timeline to span from 'first appearance' to either 'their death' or 'their final appearance'. What you'd want exactly would depend on how you structure your project; if your outline is a more classic format, appearances would be reasonable. If your outline is structured more like a chronological timeline, you might want their storyline to span from birth to death. But that would also open the door to references to characters that are actually not quite on their timeline because they have passed away. Or, in the case of some sort of prophecy plot, before their birth! Or even worse: how about a resurrection, where someone comes back alive but you'd like a part of their timeline greyed out?

I think any kind of implementation would have to do some real thinking on how to get a feature like this correct. Plots and characters may be very different concepts, but given the concept of a timeline (or storyline which isn't necessarily chronological) I do believe we would like the way both plots and characters interact with the timeline to be relatively similar to enable people to learn how to best interact with the timeline.

So with that in mind, I'll expand on my former idea, and would like to toss up an expanded syntax for references for others to think about / share their thoughts on. I do not know the nitty-gritty of the storyline implementation, so I am looking at this purely from the perspective of 'referencing with more granularity'.

## Digging into references

Currently, we have this kind of syntax:

    {P:111:Customizable description of plot \#111 that defaults to the plot name}
    {C:222:Customizable description of character\#222 that defaults to the character name}

As far as I can tell, the customizable description goes entirely unused at present. It would IMHO be nice to have those for when you hover over a reference, but that's besides the point right now.

There are a few options on how to expand this syntax.

* We can add another 'meta' field, and if so, where?
  - Before field # 1. Field # 1 currently describes the kind of entity being referenced. I feel this should stay in this spot because it is the most elementary piece of information classifying the reference.
  - Between field # 1 and # 2. Field # 2 is an ID to the entity in question. This would break the obvious relationship between the existing two fields and IMHO a bad spot.
  - **Between field # 2 and # 3. Field # 3 holds a customizable description. Logically, this spot makes sense, but in regards to backwards compatibility, the new field may end up needing to have its contents moved to the description field (which would by then become # 4)**
  - After field # 3. Depending on implementation, the most painless, or the most troublesome given that those descriptions from older versions could have used colons that now end up shoving half the description in the meta field. This would be treacherous for users to figure out, and I think the final field is something you want to have most flexibility in and least fussing on our part.

* Or we can squeeze it in with an existing field, in which case, why?
  - Field # 1 describes the kind of entity being referenced. I feel this should stay as-is because any extra information would make it look like a reference to something completely different.
  - **Field # 2 is an ID to the entity in question. A great contender, given that any 'metadata' would apply specifically to this item.**
  - Field # 3 is user-customizable. Given that any meta reference would be heavily dependent on the type of work being created, I want to leave this field completely alone for the user to use. Squeezing in extra information according to some specific format would potentially make this field a lot less flexible for users.

Having sorted through these options, the two I have bolded are IMHO preferable over the others.

## Option NEAT: Put metadata between existing fields # 2 and # 3

    {P:111:beginning:Custom Plot Description}
    {C:222::Custom Character Description}

### Pro:

* The syntax is in order of granularity. (Plot reference : specific plot id : something about that plot. Same for character.)
* No confusing 'mystery' syntax that falls out of the expectation compared to the format references are known to have.
* Complete freedom to decide the sort of syntax we want for this new field.

### Neutral:

* Any entries without special meta info (>99%) would have a double colon (::) which betrays the presence of the field (good for learning about such extended features) but might seem like a slight wart compared to what we have now.

### Con:

* Old references have their customizable description squashed into the new meta field if there is no special handling when loading files from older versions. This could probably be silently fixed when reading an older version and saving it all over again.


## Option SQUASH: Put extra metadata in fields # 2 (ID)

    {P:111-beginning:Custom Plot Description}
    {C:222:Custom Character Description}

### Pro:

* Fields do not move; no special loading/upgrading code needed for old references.

### Neutral:

* The syntax is roughly in order of granularity. (Plot reference : specific plot id - something about that plot.) but the syntax makes it unintuitive.

### Con:

* Tricky syntax. You have to know about the dash (or whatever character we use), inviting mistakes for manually entered entries. We'd likely add some sort of GUI for such meta options if a feature like this were added, but manual entry and understanding how to read a reference is still very important. It should remain simple to read at the very least.


## Final thoughts

Having spent the last hour thinking about all this (and having gotten considerably carried away), my preference goes towards **Option NEAT** as a springboard to implementing functionality like this. It would open up the gateway to more granular reference tracking so that the storyline can communicate more information in a single glance.

Examples of ideas that have come to mind for me so far:

 * Custom starting / ending points for where the coloured bands go, the meaning of which would depend on the novel.
 * Referencing specific resolution steps. This would allow a more graphical visualization of how well a plot is progressing on the storyline.
 * Annotating references to characters to separate those characters participating in the scene and those being referred to by those in the scene. These could look different on the storyline, or only show up selectively.

And no doubt a lot more depending on other uses people see for such functionality.

How do others look at this? And the options that came to mind so far? Is this worth looking into this for an early exploratory implementation, or am I just way off in my own world on how to best start improving references, and would this be a lot cause with very little future?

**Comment by Phieldar** [comment]
Amazing answer, 
I would, like you, favor the NEAT option, but I don't have the python skill to analyse it, but it sound good. Simple syntax, no confusion and as you say, it would open options. It would help me a lot in the way I use the software. I believe that more granularity is key here. I also wonder if it would  be possible to link Character to element in the world option. I believe that only 3 category of characters make it easy to be lost and confuse when you write a complex story.
Again, great suggestion and amazing understanding of my question. You reformulated it in a far better way that I did :)

**Comment by worstje** [comment]
Linking character & world would be completely out of scope of what you initially asked in this issue, and similarly to what I described which is just a syntax to make this and/or future ideas less painful to implement.

The storyline has folders & texts as its x-axis. The y-axis contains characters and/or plots. The implementation that makes it possible are the entries in the References field of folders & texts, and the syntax I was talking about applies to that field only to my (very limited) knowledge.

While I am sure some sort of linking functionality could perhaps be made possible, what is the purpose? How would you visualize it and with what purpose? For my own usage, the 'Detailed Info' tab I have on characters mostly covers my needs in terms of storing relevant information for my characters, although I guess more interaction could not hurt given how unused my World entries are in practice.

Regardless, such speculation and ideas are out of the scope for this issue which is aimed at the storyline, so you might want to write up a separate issue with your exact usecase and needs if you really need such a feature.

**Comment by Phieldar** [comment]
You are right. My second point was more a out of scope suggestion. It was misplaced for me to link it to my first proposition. I believe your reformulation about the Story Line seems clever and easier to put in place. I would love the NEAT option.

My second point was more long term need that I have while writing. A medium lenght novel could have many characters and linking them to places in the word tab would be useful. Which minor character lives in that or that village or place for exemple. In the same sense that you can link a character to a plot (in the box where you can search in a dropdown menu). But, as you said, it's way outside the mark of the suggestion of this conversation.
